packages/gatsby-theme-docs/src/utils/markdown-2-react.js

```diff
@@ -0,0 +1,12 @@
+import unified from 'unified';
+import markdown from 'remark-parse';
+import remark2react from 'remark-react';
+
+export default function markdown2React(markdownString) {
+  const result = unified()
+    .use(markdown, { commonmark: true })
+    .use(remark2react)
```

Comment:
We need to pass the option `remarkReactComponents` to map the markdown elements to our custom markdown components.
https://github.com/remarkjs/remark-react#optionsremarkreactcomponents

We can either map a few of them (e.g. link, paragraph, etc) or all of them.
In any case, we should pass the options as a second argument of the `markdown2React` function, to avoid coupling the utility function with react components.

/cc @nkuehn